test/helpers.lua

```diff
@@ -428,6 +430,34 @@ local function concat_tables(...)
   return ret
 end
 
+-- Concat map-like tables.
+--
+-- behavior: Decides what to do if a key is found in more than one map:
+--           "error": raise an error
+--           "keep":  skip
+--           "force": set the item again
+local function map_extend(behavior, ...)
+  if (behavior ~= 'error' and behavior ~= 'keep' and behavior ~= 'force') then
+    error('invalid "behavior": '..tostring(behavior))
+  end
+  local ret = {}
+  for i = 1, select('#', ...) do
+    local tbl = select(i, ...)
+    if tbl then
+      for k, v in pairs(tbl) do
+        if behavior ~= 'force' and ret[k] ~= nil then
+          if behavior == 'error' then
```

Comment:
`behaviour == 'error'` although this branch is only taken if `behaviour ~= 'force'`? Is this ever true?

Reply:
why wouldn't it be? `'force` ~= 'error'`